sdk/identity/azure-identity/Troubleshoot.md

```diff
@@ -0,0 +1,223 @@
+
+### Illegal/Invalid Argument Issues
+
+#### Client Id
+
+The Client Id is the application Id of the registered application / service principal in Azure Active Directory.
+It is a required parameter for `ClientSecretCredential` and `ClientCertificateCredential`. If you have already created your service principal
+then you can retrieve the client/app id by following the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+#### Tenant Id
+
+The tenant id is te Global Unique Identifier (GUID) that identifies your organization. It is a required parameter for
+`ClientSecretCredential` and `ClientCertificateCredential`. If you have already created your service principal
+then you can retrieve the client/app id by following the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+### Client Secret Credential Issues
+
+#### Client Secret Argument
+
+The client secret is the secret string that the application uses to prove its identity when requesting a token, this can also can be referred to as an application password.
+If you have already created a service principal you can follow the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret) to get the client secret for your application.
+
+### Client Certificate Credential Issues
+
+#### Client Certificate Argument
+
+The `Client Certificate Credential` accepts `pfx` and `pem` certificates. The certificate needs to be associated with your registered application/service principal. To create and associate a certificate with your registered app. please follow the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-1-upload-a-certificate).
+
+### Create a new service principal
+
+If you're looking to create a new service principal and would like to use that, then follow tne instructions [here](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli) to create a new service principal.
+
+## Troubleshooting Managed Identity Authentication Issues
+
+### Credential Unavailable Error
+
+#### Connection Timed Out / Connection could not be established / Target Environment could not be determined
+
+The Managed Identity credential runs only on Azure Hosted machines/servers. So ensure that you are running your application on an
+Azure Hosted resource. Currently Azure Identity SDK supports [Managed Identity Authentication](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview)
+in the below listed Azure Services, so ensure you're running your application on one of these resources and have enabled the Managed Identity on
```

Comment:
We can probably remove the first suggestion to make sure the application is running on an Azure resource, since we mention it again later on in this paragraph
```suggestion
The Managed Identity credential works only on Azure Hosted machines/servers.
Currently the Azure Identity SDK supports [Managed Identity Authentication](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview)
in the below listed Azure services; ensure you're running your application on one of these resources and have enabled the Managed Identity on
```